jnigen/test/summary_test.dart

```diff
@@ -0,0 +1,324 @@
+// Copyright (c) 2022, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// These tests validate individual characteristics in summary
+// For example, the values of methods, arguments, types, generic params etc...
+@Tags(['summarizer_test'])
+
+import 'package:jnigen/src/elements/elements.dart';
+import 'package:jnigen/src/summary/summary.dart';
+import 'package:test/test.dart';
+
+import 'test_util/summary_util.dart';
+import 'test_util/test_util.dart';
+
+const jnigenPackage = 'com.github.dart_lang.jnigen';
+const simplePackage = "$jnigenPackage.simple_package";
+
+extension MethodsForSummaryTestsOnClasses on Classes {
+  String _getSimpleName(ClassDecl c) {
+    return c.binaryName.split(".").last;
+  }
+
+  ClassDecl getClassBySimpleName(String simpleName) {
```

Comment:
Later in #364, I'll rename the methods starting with `get`, so no need to change them now.